contracts/Consensus.sol

```diff
@@ -0,0 +1,180 @@
+pragma solidity 0.4.24;
+
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+contract Consensus {
+  using SafeMath for uint256;
+
+  event InitiateChange(bytes32 indexed parentHash, address[] newSet);
+  event ChangeFinalized(address[] newSet);
+
+  struct ValidatorState {
+    // Is this a validator.
+    bool isValidator;
+    // Is a validator finalized.
+    bool isValidatorFinalized;
+    // Indexes in the currentValidators.
+    uint256[] indexes;
+  }
+
+  address public SYSTEM_ADDRESS = 0xffffFFFfFFffffffffffffffFfFFFfffFFFfFFfE;
+
+  bool public finalized = false;
+  address public owner;
+  uint256 public minStake;
+
+  address[] public currentValidators;
+  address[] public pendingValidators;
```

Comment:
`pendingValidators` sound like all the validators there are in the pending state when actually they are the validators for the next block. maybe rename in to nextValidators  or fututeValidators

Reply:
it's called `pending` on all examples/projects I've seen so far, so I kept the same convention